develop/parachain-devs/runtime-development/FRAME/add-pallet-to-the-runtime.md

```diff
@@ -0,0 +1,122 @@
+
+!!! note
+    For information about adding dependencies in `Cargo.toml` files, see [Dependencies](https://doc.rust-lang.org/cargo/guide/dependencies.html){target=\_blank} page in the Cargo documentation. For information about enabling and managing features from dependent packages, see [Features](https://doc.rust-lang.org/cargo/reference/features.html){target=\_blank} section in the Cargo documentation.
+
+## Dependencies for a New Pallet
+
+To add the dependencies for a new pallet to the runtime, you need to modify the `Cargo.toml` file by adding a new line into the `[workspace.dependencies]` section with the pallet you want to add. Ensure that you stick to the following format:
+
+```toml
+pallet-example = { version = "4.0.0-dev", default-features = false }
+```
+
+This line imports the pallet-example crate as a dependency and specifies the following:
+
+- `version` - the specific version of the crate to import
+- `default-features` - determines the behavior for including pallet features when compiling the runtime with standard Rust libraries
+
+Next, add this dependency to the `[dependencies]` section of the `runtime/Cargo.toml` file, so it inherits from the main `Cargo.toml` file:
+
+```toml
+pallet-indices.workspace = true
+```
+
+To enable the `std` feature of the pallet, add the pallet to the following section:
+
+```toml
+[features]
+default = ["std"]
+std = [
+    ...
+    "pallet-example/std",
+    ...
+]
+```
+
+This section specifies the default feature set for the runtime, which includes the `std` features for each pallet. When the runtime is compiled with the `std` feature set, the standard library features for all listed pallets are enabled. For more details on how the runtime is compiled as both a native binary (using `std`) and a Wasm binary (using `no_std`), refer to the [Wasm build](https://paritytech.github.io/polkadot-sdk/master/polkadot_sdk_docs/polkadot_sdk/substrate/index.html#wasm-build){target=\_blank} section in the Polkadot SDK documentation.
+
+!!! note
+    If you forget to update the features section in the `Cargo.toml` file, you might encounter `cannot find function` errors when compiling the runtime.
+
+To ensure that the new dependencies resolve correctly for the runtime, you can run the following command:
+
+```bash
+cargo check -p solochain-template-runtime --release
+```
+
+## Config Trait for Pallets
+
+Every Polkadot SDK pallet defines a [Rust trait](https://doc.rust-lang.org/book/ch10-02-traits.html){target=\_blank} called `Config`. This trait specifies the types and parameters that the pallet needs to integrate with the runtime and perform its functions. The primary purpose of this trait is to act as an interface between this pallet and the runtime in which it is embedded. A type, function, or constant in this trait is essentially left to be configured by the runtime that includes this pallet.
```

Comment:
We should reference the actual `Config` trait from the source code and not the definition of a rust trait

Reply:
Mmm, I received some comments from @eshaben in some old PRs regarding this topic. She stated that it's preferable to refer to the Rust docs instead of the source code because the source code might change in the future, and that would make our docs hard to maintain

Reply:
I understand it for this case because the `Config` explanation is kind of generic